src/app/(public)/repos/_components/pagination.tsx

```diff
@@ -15,24 +18,69 @@ export function Pagination({
   totalCount,
   searchParams
 }: PaginationProps) {
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+
+  const handlePrevPage = (e: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
I prefer using named functions instead of arrow functions when defining them

```suggestion
  function handlePrevPage(e: React.MouseEvent<HTMLButtonElement>) {
```